openwisp_notifications/handlers.py

```diff
@@ -274,16 +274,58 @@ def notification_type_registered_unregistered_handler(sender, **kwargs):
 
 
 @receiver(
-    pre_save,
+    post_save,
     sender=OrganizationUser,
     dispatch_uid='create_orguser_notification_setting',
 )
-def notification_setting_org_user_created(instance, **kwargs):
-    if instance.is_admin:
-        tasks.ns_organization_user_added_or_updated.delay(
-            instance_id=instance.pk,
-            instance_user_id=instance.user_id,
-            instance_org_id=instance.organization_id,
+def notification_setting_org_user_created(instance, created, **kwargs):
+    if created and instance.is_admin:
+        tasks.create_notification_settings_for_org_user.delay(
+            org_user_id=instance.pk,
+            user_id=instance.user_id,
+            org_id=instance.organization_id,
+        )
+
+
+@receiver(
+    pre_save,
+    sender=OrganizationUser,
+    dispatch_uid='update_orguser_notification_setting',
+)
+def notification_setting_org_user_org_updated(instance, **kwargs):
```

Comment:
This is very rudimentary code which have further scope to be improved(optimised). I wrote it to realise the bug and to explicitly share the corner cases which were missing in the previous implementation. 

I have also tried to take a different approach that does not require these much computation or code. I opened https://github.com/pandafy/openwisp-notifications/pull/4 for proof of concept.